packages/xdl/src/Simulator.js

```diff
@@ -214,6 +216,13 @@ async function _listSimulatorDevicesAsync() {
   return info;
 }
 
+async function _waitForDeviceToBoot() {
+  let bootedDevice = await _bootedSimulatorDeviceAsync();
+  while (!bootedDevice) {
+    bootedDevice = await _bootedSimulatorDeviceAsync();
+  }
```

Comment:
A `do-while` loop would be nice here.

I think you might also want to add a small wait between the tries (e.g. `await delayAsync(100)`) in this loop so we don't continuously fire the `xcrun simctl list devices` command.

Also what happens if the device never boots? Perhaps add a timeout/counter to make sure we won't be stuck in this loop indefinitely.

Reply:
added a timeout, set for 10 seconds. When testing, it never took longer than 5, but let me know if you think it should be set to longer/shorter 👍